AGENTS.md

```diff
@@ -0,0 +1,202 @@
+```
+
+## Working with Jsonnet
+
+### Updating Jsonnet Dependencies
+When updating vendored jsonnet libraries (e.g., kube-prometheus):
+
+```bash
+cd jsonnet
+# Edit jsonnetfile.json to update version
+jb update
+cd ..
+make generate           # Regenerate all assets
+```
+
+See `Documentation/development.md` for detailed workflow on updating individual jsonnet bundles.
+
+### Modifying Manifests
+For rapid iteration on manifest changes:
+
+1. Build operator: `make operator`
+2. Modify YAML files in `assets/` directory directly
+3. Run operator: `./hack/local-cmo.sh`
+4. Test changes
+5. Port changes back to `jsonnet/` source files
+6. Run `make generate` to regenerate assets from jsonnet
+
+**Important**: Some resources require a two-release process for removing labels/annotations:
+- First release: Add suffix `"-"` to the annotation/label (CMO will delete it)
+- Second release: Remove the annotation/label from the manifest
+
+## Key Patterns
+
+### Component Reconciliation
+Each component task follows a standard pattern:
+1. Load manifests from `assets/` directory
+2. Apply configuration from ConfigMap
+3. Create or update Kubernetes resources
+4. Handle resource dependencies and ordering
+
+### Configuration Validation
+Configuration is validated in `pkg/manifests/` using:
+- Struct tags for basic validation
+- Custom validation functions for complex rules
+- Integration with OpenShift FeatureGates for feature availability
+
+### Resource Management
+The operator uses library-go patterns for:
+- Certificate rotation (`certrotation`)
+- CSR approval for node-exporter TLS
+- Event recording and status reporting
+- Resource merging and updates
+
+## Testing Strategy
+
+### Unit Tests
+- Located alongside source files (`*_test.go`)
+- Test individual functions and components
+- Mock Kubernetes clients where needed
+
+### E2E Tests
+- Located in `test/e2e/`
+- Test operator behavior against real cluster
+- Require KUBECONFIG environment variable
+- Cover configuration changes, upgrades, and component interactions
+
+### Rule Tests
+- Located in `test/rules/`
+- Validate Prometheus alerting and recording rules
+- Check rule syntax and runbook URLs
+
+## Adding New Monitoring Components
+
+To add a new monitoring component:
+
+1. Create jsonnet definition in `jsonnet/components/<component>.libsonnet`
+2. Add component to `jsonnet/main.jsonnet`
+3. Create task implementation in `pkg/tasks/<component>.go`
+4. Register task in `pkg/operator/operator.go` reconciliation loop
+5. Add configuration types to `pkg/manifests/types.go` if needed
+6. Run `make generate` to generate manifests
+7. Add e2e tests in `test/e2e/`
+8. Update documentation
+
+## Contributing
+
+- All PRs must reference a JIRA ticket in the title: `OCPBUGS-12345: ...` or `MON-1234: ...`
+- Follow commit message format: `<subsystem>: <what changed>`
+- Ensure `make verify` passes before submitting PR
+- Add tests for new functionality
+- Update documentation as needed
```

Comment:
This just repeats what is noted in `CONTRIBUTING.md`.

It would be more valuable for example to outline how an agent needs to interact with Jira.